### PR TITLE
Added Status method and healthcheck http server

### PR DESCRIPTION
**Pull request by Gobonoid**

### CODE REVIEW

File: proximo-server/amqp.go [hunk 1]
```diff
@@ -102,3 +103,18 @@ func (h *amqpHandler) HandleConsume(ctx context.Context, consumer, topic string,
 func (h *amqpHandler) HandleProduce(ctx context.Context, topic string, forClient chan<- *Confirmation, messages <-chan *Message) error {
 	panic("not implemented")
 }
+
+func (h *amqpHandler) Status() (bool, []error) {
```
Comment (ggordan) [inline proximo-server/amqp.go]: I'm not really a fan of returning an array of errors. Would it be worth creating a new error type instead and just changing this to return `(bool, error)`

Comment (Gobonoid) [reply proximo-server/amqp.go]: @ggordan I supose we can do it :)

File: proximo-server/kafka.go [hunk 2]
```diff
@@ -143,3 +145,34 @@ func (h *kafkaHandler) HandleProduce(ctx context.Context, topic string, forClien
 		}
 	}
 }
+
+func (h *kafkaHandler) Status() (bool, []error) {
+	errs := make(chan error)
```
Comment (ggordan) [inline proximo-server/kafka.go]: Is it worth making this a buffered channel (number of brokers) such that we're not blocked when reading

File: proximo-server/kafka.go [hunk 3]
```diff
@@ -143,3 +145,34 @@ func (h *kafkaHandler) HandleProduce(ctx context.Context, topic string, forClien
...
+				errs <- fmt.Errorf("Failed to close connection to broker %s: %v", broker, err)
+				return
+			}
+			errs <- nil
+		}(broker)
+	}
+
+	errors := []error{}
+	for range h.brokers {
+		err := <-errs
```
Comment (ggordan) [inline proximo-server/kafka.go]: It's probably worth doing a `select` in here so we can time out if we don't get a response in a reasonable time

File: proximo-server/kafka.go (hunk 2 above)
Comment (Gobonoid) [reply proximo-server/kafka.go]: @ggordan If I be honest, I copied this method from other ops project. Didn't want to do any magic about it here.

File: proximo-server/kafka.go (hunk 3 above)
Comment (Gobonoid) [reply proximo-server/kafka.go]: @ggordan line 154 implements timeout in connection.

**Comment by mjgarton** [comment]
Most of this looks familiar.  Is it based on the go-pubsub status code?

This PR LGTM, but I wonder what you think of us adding (in future) the core concept of proximo status (Statuser +  impls) into the uw-labs version, possibly exposing it somehow via additions to the proximo protocol, and keeping the uw specific http HC endpoint here obviously using the same shared code to implement it?

**Comment by Gobonoid** [comment]
@mjgarton That's correct. I've used some code from go-pubsub for kafka I think.

TBH I was thinking about it when I was doing this work, but things around ops were a bit specific here (all those apps descriptions, names etc). I think we can spend more time on this in the future. Definitely not a waste of time.

**Pull request merged**